Cargo.toml

```diff
@@ -17,28 +17,27 @@ travis-ci = { repository = "gimli-rs/gimli" }
 coveralls = { repository = "gimli-rs/gimli" }
 
 [dependencies]
-arrayvec = { version = "0.5.0", default-features = false }
+arrayvec = { version = "0.5.0", default-features = false, optional = true }
 byteorder = { version = "1.0", default-features = false }
-fallible-iterator = { version = "0.2.0", default-features = false }
+fallible-iterator = { version = "0.2.0", default-features = false, optional = true }
 indexmap = { version = "1.0.2", optional = true }
-smallvec = { version = "0.6.10", default-features = false }
+smallvec = { version = "1.1.0", default-features = false, optional = true }
 stable_deref_trait = { version = "1.1.0", default-features = false, optional = true }
 
 [dev-dependencies]
 crossbeam = "0.7.1"
 getopts = "0.2"
 memmap = "0.7"
 num_cpus = "1"
-object = "0.15"
+object = "0.17"
 rayon = "1.0"
 regex = "1"
 test-assembler = "0.1.3"
-typed-arena = "1"
-target-lexicon = "0.8"
+typed-arena = "2"
 
 [features]
-read = ["stable_deref_trait"]
-write = ["std", "indexmap"]
+read = ["arrayvec", "fallible-iterator", "smallvec", "stable_deref_trait"]
+write = ["indexmap"]
 std = ["fallible-iterator/std", "stable_deref_trait/std"]
```

Comment:
I think this will cause `fallible-iterators` to be enabled when `std` is enabled but `read` is disabled.

Reply:
Yes, as in the other comment, I don't know how to enable `fallible-iterators/std` only if both `read` and `std` are enabled. So my workaround is to not enable `std` for `write`.